internal/integration/integration_test.go

```diff
@@ -0,0 +1,124 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Package integration_test contains core runners for integration tests
+package integration_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/suite"
+	"gopkg.in/yaml.v3"
+	capiv1 "sigs.k8s.io/cluster-api/cmd/clusterctl/api/v1alpha3"
+
+	"github.com/talos-systems/capi-utils/pkg/capi"
+	"github.com/talos-systems/capi-utils/pkg/capi/infrastructure"
+)
+
+type clusterctlConfig struct {
+	Providers []controlplaneProvider `yaml:"providers"`
+}
+
+type controlplaneProvider struct {
+	Name         string              `yaml:"name"`
+	Url          string              `yaml:"url"`
+	ProviderType capiv1.ProviderType `yaml:"type"`
+}
+
+type IntegrationSuite struct {
+	suite.Suite
+
+	manager *capi.Manager
+	cluster *capi.Cluster
+}
+
+func (suite *IntegrationSuite) TearDownSuite() {
+	if suite.cluster != nil {
+		err := suite.manager.DestroyCluster(context.Background(), suite.cluster.Name(), suite.cluster.Namespace(), suite.cluster.CAPIVersion())
+		suite.Require().NoError(err)
+	}
+}
+
+// TestIntegration runs integration tests.
+func (suite *IntegrationSuite) TestIntegrationAWS() {
+	env := func(key, def string) string {
+		val := os.Getenv(key)
+		if val != "" {
+			return val
+		}
+
+		return def
+	}
+
+	providerType := env("PROVIDER", "aws:v0.6.7")
```

Comment:
should this be part of the `SetupSuite()` ? as we should do it once for all the tests I guess